api/v1beta1/types_class.go

```diff
@@ -459,6 +459,10 @@ type NetworkClassSpec struct {
 	// +optional
 	PrivateDNSZoneName string `json:"privateDNSZoneName,omitempty"`
 
+	// PrivateDNSZone enables private dns zone creation modes for private cluster.
+	// +optional
+	PrivateDNSZone *string `json:"privateDNSZone,omitempty"`
```

Comment:
It should probably be noted in the API that if not specified, this defaults to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/71bd7f15f7345aeb3872a86a6e2e0a2034b89392/api/v1beta1/azuremanagedcontrolplane_types.go#L30.

Reply:
Done. Thanks for taking a look.